Src/LiquidProjections/Statistics/ProjectorStats.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+
+namespace LiquidProjections.Statistics
+{
+    /// <summary>
+    /// Contains statistics and information about a particular projector.
+    /// </summary>
+    public class ProjectorStats
+    {
+        private readonly ConcurrentDictionary<string, Property> properties =
+            new ConcurrentDictionary<string, Property>();
+
+        private readonly BlockingCollection<Event> events = new BlockingCollection<Event>();
+
+        private readonly WeightedProjectionSpeedCalculator lastMinuteSamples =
+            new WeightedProjectionSpeedCalculator(12, TimeSpan.FromSeconds(5));
+
+        private readonly WeightedProjectionSpeedCalculator last10MinuteSamples
+            = new WeightedProjectionSpeedCalculator(9, TimeSpan.FromMinutes(1));
+
+        public ProjectorStats(string projectorId)
+        {
+            ProjectorId = projectorId;
+        }
+
+        public ProjectorStats(string projectorId, IDictionary<string, Property> properties, IEnumerable<Event> events)
+        {
+            this.properties = new ConcurrentDictionary<string, Property>(properties);
+
+            foreach (Event @event in events)
+            {
+                this.events.Add(@event);
+            }
+
+            ProjectorId = projectorId;
+        }
+
+        public string ProjectorId { get; }
+
+        public TimestampedCheckpoint LastCheckpoint { get; set; }
+
+        public IDictionary<string, Property> Properties => properties;

```

Comment:
I don't recommend exposing `ConcurrentDictionary` as `IDictionary` here. Each single operation on it is thread-safe but a series of operations may see the collection changed in-between them. For example, enumeration may fail. We can have a method that copies the property to a new collection instead.